### ISSUE TITLE
Missing a couple small styles

### ISSUE THREAD

**Issue by zachthedev** [rakibdev/material-code#6]
Thank you so much for this!

After setting my environment, I noticed there seemed to be several very small unstyled parts:

- activityBarBadge.background
- banner.iconForeground
- inputOption.activeBorder
- pickerGroup.foreground
- textLink.activeForeground
- notificationsInfoIcon.foreground

Edit: Here is another -
- problemsInfoIcon.foreground

I also ported the editor token color styles from vscode Material Theme, as I felt it fit more than the default ones you have. Just in case you are interested:

<details>
  <summary>snippet from settings.js</summary>

  ```json
  "editor.tokenColorCustomizations": {
    "[Material Code]": {
      "textMateRules": [
        {
          "name": "String",
          "scope": "string",
          "settings": {
            "foreground": "#c3e88d"
          }
        },
        {
          "name": "Punctuation",
          "scope": "punctuation, constant.other.symbol",
          "settings": {
            "foreground": "#89ddff"
          }
        },
        {
          "name": "String Escape",
          "scope": "constant.character.escape, text.html constant.character.entity.named",
          "settings": {
            "foreground": "#ede0dc"
          }
        },
        {
          "name": "Boolean",
          "scope": "constant.language.boolean",
          "settings": {
            "foreground": "#ff9cac"
          }
        },
        {
          "name": "Number",
          "scope": "constant.numeric",
          "settings": {
            "foreground": "#f78c6c"
          }
        },
        {
          "name": "Variable",
          "scope": "variable, variable.parameter, support.variable, variable.language, variable.other, support.constant, meta.definition.variable entity.name.function, meta.function-call.arguments",
          "settings": {
            "foreground": "#ede0dc"
          }
        },
        {
          "name": "Other Keyword",
          "scope": "keyword.other",
          "settings": {
            "foreground": "#f78c6c"
          }
        },
        {
          "name": "Keyword",
          "scope": "keyword, keyword.operator, modifier, variable.language.this, support.type.object, constant.language",
          "settings": {
            "foreground": "#89ddff"
          }
        },
        {
          "name": "Function call",
          "scope": "entity.name.function, support.function",
          "settings": {
            "foreground": "#82aaff"
          }
        },
        {
          "name": "Storage",
          "scope": "storage.type, storage.modifier, storage.control",
          "settings": {
            "foreground": "#c792ea"
          }
        },
        {
          "name": "Modules",
          "scope": "support.module, support.node",
          "settings": {
            "fontStyle": "italic",
            "foreground": "#f07178"
          }
        },
        {
          "name": "Type",
          "scope": "support.type, constant.other.key",
          "settings": {
            "foreground": "#ffcb6b"
          }
        },
        {
          "name": "Type",
          "scope": "entity.name.type, entity.other.inherited-class, entity.other",
          "settings": {
            "foreground": "#ffcb6b"
          }
        },
        {
          "name": "Comment",
          "scope": "comment",
          "settings": {
            "fontStyle": "italic",
            "foreground": "#545454"
          }
        },
        {
          "name": "Comment",
          "scope": "comment punctuation.definition.comment, string.quoted.docstring",
          "settings": {
            "fontStyle": "italic",
            "foreground": "#545454"
          }
        },
        {
          "name": "Punctuation",
          "scope": "punctuation",
          "settings": {
            "foreground": "#89ddff"
          }
        },
        {
          "name": "Class",
          "scope": "entity.name, entity.name.type.class, support.type, support.class, meta.use",
          "settings": {
            "foreground": "#ffcb6b"
          }
        },
        {
          "name": "Class variable",
          "scope": "variable.object.property, meta.field.declaration entity.name.function",
          "settings": {
            "foreground": "#f07178"
          }
        },
        {
          "name": "Class method",
          "scope": "meta.definition.method entity.name.function",
          "settings": {
            "foreground": "#f07178"
          }
        },
        {
          "name": "Function definition",
          "scope": "meta.function entity.name.function",
          "settings": {
            "foreground": "#82aaff"
          }
        },
        {
          "name": "Template expression",
          "scope": "template.expression.begin, template.expression.end, punctuation.definition.template-expression.begin, punctuation.definition.template-expression.end",
          "settings": {
            "foreground": "#89ddff"
          }
        },
        {
          "name": "Reset embedded/template expression colors",
          "scope": "meta.embedded, source.groovy.embedded, meta.template.expression",
          "settings": {
            "foreground": "#ede0dc"
          }
        },
        {
          "name": "YAML key",
          "scope": "entity.name.tag.yaml",
          "settings": {
            "foreground": "#f07178"
          }
        },
        {
          "name": "JSON key",
          "scope": "meta.object-literal.key, meta.object-literal.key string, support.type.property-name.json",
          "settings": {
            "foreground": "#f07178"
          }
        },
        {
          "name": "JSON constant",
          "scope": "constant.language.json",
          "settings": {
            "foreground": "#89ddff"
          }
        },
        {
          "name": "CSS class",
          "scope": "entity.other.attribute-name.class",
          "settings": {
            "foreground": "#ffcb6b"
          }
        },
        {
          "name": "CSS ID",
          "scope": "entity.other.attribute-name.id",
          "settings": {
            "foreground": "#f78c6c"
          }
        },
        {
          "name": "CSS tag",
          "scope": "source.css entity.name.tag",
          "settings": {
            "foreground": "#ffcb6b"
          }
        },
        {
          "name": "CSS properties",
          "scope": "support.type.property-name.css",
          "settings": {
            "foreground": "#b2ccd6"
          }
        },
        {
          "name": "HTML tag outer",
          "scope": "meta.tag, punctuation.definition.tag",
          "settings": {
            "foreground": "#89ddff"
          }
        },
        {
          "name": "HTML tag inner",
          "scope": "entity.name.tag",
          "settings": {
            "foreground": "#f07178"
          }
        },
        {
          "name": "HTML tag attribute",
          "scope": "entity.other.attribute-name",
          "settings": {
            "foreground": "#c792ea"
          }
        },
        {
          "name": "HTML entities",
          "scope": "punctuation.definition.entity.html",
          "settings": {
            "foreground": "#ede0dc"
          }
        },
        {
          "name": "Markdown heading",
          "scope": "markup.heading",
          "settings": {
            "foreground": "#89ddff"
          }
        },
        {
          "name": "Markdown link text",
          "scope": "text.html.markdown meta.link.inline, meta.link.reference",
          "settings": {
            "foreground": "#f07178"
          }
        },
        {
          "name": "Markdown list item",
          "scope": "text.html.markdown beginning.punctuation.definition.list",
          "settings": {
            "foreground": "#89ddff"
          }
        },
        {
          "name": "Markdown italic",
          "scope": "markup.italic",
          "settings": {
            "fontStyle": "italic",
            "foreground": "#f07178"
          }
        },
        {
          "name": "Markdown bold",
          "scope": "markup.bold",
          "settings": {
            "fontStyle": "bold",
            "foreground": "#f07178"
          }
        },
        {
          "name": "Markdown bold italic",
          "scope": "markup.bold markup.italic, markup.italic markup.bold",
          "settings": {
            "fontStyle": "italic bold",
            "foreground": "#f07178"
          }
        },
        {
          "name": "Markdown code block",
          "scope": "markup.fenced_code.block.markdown punctuation.definition.markdown",
          "settings": {
            "foreground": "#c3e88d"
          }
        },
        {
          "name": "Markdown inline code",
          "scope": "markup.inline.raw.string.markdown",
          "settings": {
            "foreground": "#c3e88d"
          }
        },
        {
          "name": "INI property name",
          "scope": "keyword.other.definition.ini",
          "settings": {
            "foreground": "#f07178"
          }
        },
        {
          "name": "INI section title",
          "scope": "entity.name.section.group-title.ini",
          "settings": {
            "foreground": "#89ddff"
          }
        },
        {
          "name": "C# class",
          "scope": "source.cs meta.class.identifier storage.type",
          "settings": {
            "foreground": "#ffcb6b"
          }
        },
        {
          "name": "C# class method",
          "scope": "source.cs meta.method.identifier entity.name.function",
          "settings": {
            "foreground": "#f07178"
          }
        },
        {
          "name": "C# function call",
          "scope": "source.cs meta.method-call meta.method, source.cs entity.name.function",
          "settings": {
            "foreground": "#82aaff"
          }
        },
        {
          "name": "C# type",
          "scope": "source.cs storage.type",
          "settings": {
            "foreground": "#ffcb6b"
          }
        },
        {
          "name": "C# return type",
          "scope": "source.cs meta.method.return-type",
          "settings": {
            "foreground": "#ffcb6b"
          }
        },
        {
          "name": "C# preprocessor",
          "scope": "source.cs meta.preprocessor",
          "settings": {
            "foreground": "#545454"
          }
        },
        {
          "name": "C# namespace",
          "scope": "source.cs entity.name.type.namespace",
          "settings": {
            "foreground": "#ede0dc"
          }
        },
        {
          "name": "JSX Text",
          "scope": "meta.jsx.children, SXNested",
          "settings": {
            "foreground": "#ede0dc"
          }
        },
        {
          "name": "JSX Components name",
          "scope": "support.class.component",
          "settings": {
            "foreground": "#ffcb6b"
          }
        },
        {
          "name": "C-related Block Level Variables",
          "scope": "source.cpp meta.block variable.other",
          "settings": {
            "foreground": "#ede0dc"
          }
        },
        {
          "name": "Member Access Meta",
          "scope": "source.python meta.member.access.python",
          "settings": {
            "foreground": "#f07178"
          }
        },
        {
          "name": "Function Call",
          "scope": "source.python meta.function-call.python, meta.function-call.arguments",
          "settings": {
            "foreground": "#82aaff"
          }
        },
        {
          "name": "Blocks",
          "scope": "meta.block",
          "settings": {
            "foreground": "#f07178"
          }
        },
        {
          "name": "Function Call",
          "scope": "entity.name.function.call",
          "settings": {
            "foreground": "#82aaff"
          }
        },
        {
          "name": "Namespaces",
          "scope": "source.php support.other.namespace, source.php meta.use support.class",
          "settings": {
            "foreground": "#ede0dc"
          }
        },
        {
          "name": "Constant keywords",
          "scope": "constant.keyword",
          "settings": {
            "fontStyle": "italic",
            "foreground": "#89ddff"
          }
        },
        {
          "name": "Entity name",
          "scope": "entity.name.function",
          "settings": {
            "foreground": "#82aaff"
          }
        },
        {
          "name": "Constant Placeholder",
          "scope": ["constant.other.placeholder"],
          "settings": {
            "foreground": "#f07178"
          }
        },
        {
          "name": "Markup Deleted",
          "scope": ["markup.deleted"],
          "settings": {
            "foreground": "#f07178"
          }
        },
        {
          "name": "Markup Inserted",
          "scope": ["markup.inserted"],
          "settings": {
            "foreground": "#c3e88d"
          }
        },
        {
          "name": "Markup Underline",
          "scope": ["markup.underline"],
          "settings": {
            "fontStyle": "underline"
          }
        },
        {
          "name": "Keyword Control",
          "scope": ["keyword.control"],
          "settings": {
            "fontStyle": "italic",
            "foreground": "#89ddff"
          }
        },
        {
          "name": "Parameter",
          "scope": ["variable.parameter"],
          "settings": {
            "fontStyle": "italic"
          }
        },
        {
          "name": "Python - Self Parameter",
          "scope": ["variable.parameter.function.language.special.self.python"],
          "settings": {
            "fontStyle": "italic",
            "foreground": "#f07178"
          }
        },
        {
          "name": "Python - Format Placeholder",
          "scope": ["constant.character.format.placeholder.other.python"],
          "settings": {
            "foreground": "#f78c6c"
          }
        },
        {
          "name": "Markdown - Blockquote",
          "scope": ["markup.quote"],
          "settings": {
            "fontStyle": "italic",
            "foreground": "#89ddff"
          }
        },
        {
          "name": "Markdown - Fenced Language",
          "scope": ["markup.fenced_code.block"],
          "settings": {
            "foreground": "#ede0dc90"
          }
        },
        {
          "name": "Markdown - Blockquote Punctuation",
          "scope": ["punctuation.definition.quote"],
          "settings": {
            "foreground": "#ff9cac"
          }
        },
        {
          "name": "JSON Key - Level 0",
          "scope": ["meta.structure.dictionary.json support.type.property-name.json"],
          "settings": {
            "foreground": "#c792ea"
          }
        },
        {
          "name": "JSON Key - Level 1",
          "scope": [
            "meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json support.type.property-name.json"
          ],
          "settings": {
            "foreground": "#ffcb6b"
          }
        },
        {
          "name": "JSON Key - Level 2",
          "scope": [
            "meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json support.type.property-name.json"
          ],
          "settings": {
            "foreground": "#f78c6c"
          }
        },
        {
          "name": "JSON Key - Level 3",
          "scope": [
            "meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json support.type.property-name.json"
          ],
          "settings": {
            "foreground": "#f07178"
          }
        },
        {
          "name": "JSON Key - Level 4",
          "scope": [
            "meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json support.type.property-name.json"
          ],
          "settings": {
            "foreground": "#b98ebc"
          }
        },
        {
          "name": "JSON Key - Level 5",
          "scope": [
            "meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json support.type.property-name.json"
          ],
          "settings": {
            "foreground": "#82aaff"
          }
        },
        {
          "name": "JSON Key - Level 6",
          "scope": [
            "meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json support.type.property-name.json"
          ],
          "settings": {
            "foreground": "#ff9cac"
          }
        },
        {
          "name": "JSON Key - Level 7",
          "scope": [
            "meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json support.type.property-name.json"
          ],
          "settings": {
            "foreground": "#c792ea"
          }
        },
        {
          "name": "JSON Key - Level 8",
          "scope": [
            "meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json meta.structure.dictionary.value.json meta.structure.dictionary.json support.type.property-name.json"
          ],
          "settings": {
            "foreground": "#c3e88d"
          }
        },
        {
          "scope": ["", "keyword", "keyword.other.unit", "constant.other", "comment", "punctuation.definition.comment"],
          "settings": {
            "foreground": "#baa099"
          }
        },
        {
          "scope": [""],
          "settings": {
            "foreground": "#ede0dc"
          }
        }
      ]
    }
  },
  ```
</details>

**Comment by rakibdev** [comment]
@zachthedev 
good catch! added https://github.com/rakibdev/material-code/commit/956524329a40772d1fc1dc020287f5bd01ac6890

> ported the editor token color styles from vscode Material Theme

thinking of doing similar. like https://github.com/microsoft/vscode/issues/25986 adding a settings input option "Use different syntax theme" that'll automatically extract user's preferred theme token colors and use it. leaving the field empty will use default Material Code's token colors.

**Comment by zachthedev** [comment]
@rakibdev No problem! Thanks for adding them.

Yeah, that would be perfect! Just letting you know that the snippet that I provided isn't perfect, I am noticing that there are some styles still shown from this repo that I need to figure out but still gets my idea across.

**Comment by rakibdev** [comment]
@zachthedev 

> some styles still shown from this repo

you mean token colors didn't affect bracket or pair guide colors? this fixed. here's how it looks

https://github.com/rakibdev/material-code/assets/60188988/f4ba60cc-bddf-4d0a-b1bb-a864935163dd

**Comment by rakibdev** [comment]
now available in [v2.6.0](https://github.com/rakibdev/material-code/releases/tag/2.6.0)

**Comment by RVFET** [comment]
I wonder if its possible to automatically extract another theme's syntax theme. Meaning, if a theme does not provide a separate syntaxtheme, could we somehow use the said theme as syntaxtheme still ? I know this sentence is pretty wild, but I hope you understood what I meant.

![Image](https://github.com/user-attachments/assets/045dcf3a-0968-44ea-ae08-c09b6880b2df)

For example, I'd like to use `Bikkuri (Dark Moss & Lively)` as syntax theme, but it is not available.

**Comment by rakibdev** [comment]
@RVFET 
"Dark Modern" is a example, using any installed theme name should work.
In v3.0.5 this is example is replaced with a syntax theme picker to avoid confusion https://github.com/rakibdev/material-code/releases

Tried "Bikkuri (Dark Moss & Lively)" still didn't work. JSON.parse was failing due to Bikkuri having inline comments in JSON, this is also fixed in v3.0.5.

https://github.com/user-attachments/assets/41f9d1ec-d47a-45bf-85f0-f715ca4295ba
